ecoscope/io/earthranger.py

```diff
@@ -694,6 +695,36 @@ def get_patrols(self, since=None, until=None, patrol_type=None, status=None, **a
             df = df.sort_values(by="serial_number").reset_index(drop=True)
         return df
 
+    def get_patrol_events(self, since=None, until=None, patrol_type=None, status=None, **addl_kwargs):
+        """
+        Parameters
+        ----------
+        since:
+            lower date range
+        until:
+            upper date range
+        patrol_type:
+            Comma-separated list of type of patrol UUID
+        status
+            Comma-separated list of 'scheduled'/'active'/'overdue'/'done'/'cancelled'
+        Returns
+        -------
+        events : pd.DataFrame
+            DataFrame of queried patrols
+        """
+        patrol_df = self.get_patrols(since=since, until=until, patrol_type=patrol_type, status=status, **addl_kwargs)
+
+        events = []
+        for _, row in patrol_df.iterrows():
+            if row["patrol_segments"]:
+                for segment in row["patrol_segments"]:
+                    for event in segment.get("events", []):
+                        events.append(event)
```

Comment:
Is it worth adding the patrol id / segment id into the returned data?
```suggestion
                        event["patrol_id"] = row.get("id")
                        event["patrol_segment_id"] = segment.get("id")
                        events.append(event)
```